### PR TITLE
Issue/7709 whats next sites

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add a ***What's Next*** list item to the ***Sites*** tab as described in https://github.com/wordpress-mobile/WordPress-Android/issues/7709.  This ***does not*** include triggering ***What's Next*** prompts when the user visits an associated screen as originally planned.  Those triggers are dependent upon the checklist progress determined in a subsequent issue.  The only trigger included is the initial ***What's Next*** dialog shown when creating a new site.

There are multiple differences in the interface since the original design.  See the descriptions and screenshots below for details.
- What's Next was changed to Quick Start
- Prompt for help changed from [`WPDialogSnackbar`](https://github.com/wordpress-mobile/WordPress-Android/blob/feature/whats-next/WordPress/src/main/java/org/wordpress/android/widgets/WPDialogSnackbar.java) to [`PromoDialog`](https://github.com/wordpress-mobile/WordPress-Android/blob/feature/whats-next/WordPress/src/main/java/org/wordpress/android/ui/posts/PromoDialog.kt)
- Progress (i.e. 1/7) text color and size updated to match style of other views (i.e. Plan)
- Total number shown in progress is 7 to match the number of tasks
- Never button added to prompt asking for help
- Copy updated after Editorial review

sites_permission

sites_progress

sites_completed

#### Note
The ***Quick Start*** view is invisible.  It will become visible in a subsequent issue.

### Test
1. Go to ***Sites*** tab.
2. Tap ***Switch Site*** button.
3. Tap ***+*** action.
4. Tap ***Create WordPress.com site*** option.
5. Go through creating site.
6. Tap ***Configure*** button.
7. Notice ***Quick Start*** dialog is shown.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 
 import org.greenrobot.eventbus.Subscribe;
 import org.greenrobot.eventbus.ThreadMode;
+import org.jetbrains.annotations.NotNull;
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java]: Do we really need to use this Jetbrains annotation when we're already using `android.support.annotation.NonNull` in this fragment?

File: WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java [hunk 2]
```diff
@@ -209,6 +209,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             case RequestCodes.CREATE_SITE:
                 if (resultCode == RESULT_OK) {
                     debounceLoadSites();
+
+                    if (data != null) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java]: If `data` is null should we create it here? Looking at `SiteCreationActivity` I see [one situation](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/accounts/SiteCreationActivity.java#L148) where the result with be `RESULT_OK` and `data` will be null even if a site has been created.

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 3]
```diff
@@ -527,9 +534,14 @@ private void trackLastVisiblePage(int position, boolean trackAnalytics) {
             case PAGE_MY_SITE:
                 ActivityId.trackLastActivity(ActivityId.MY_SITE);
                 if (trackAnalytics) {
-                    AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.MY_SITE_ACCESSED,
-                                                        getSelectedSite());
+                    AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.MY_SITE_ACCESSED, getSelectedSite());
                 }
+
+                MySiteFragment fragment = getMySiteFragment();
+                if (fragment != null) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: Is it possible to keep the quick start code in the activity? It seems a shame to have it spread between the activity and the fragment, but perhaps that's necessary for future PRs?

**Comment by nbradbury** [comment]
Looks good! :shipit:

**Comment by khaykov** [comment]
Part of #7707

**Pull request merged**